app/javascript/react/ansibleCatalog/catalogActions.js

```diff
@@ -0,0 +1,84 @@
+import * as actionTypes from './actionTypes';
+import { API } from '../../http_api';
+
+export const commonParams = '&expand=resources&sort_by=name&sort_order=ascending';
+export const credentialsRequest = type => `/api/authentications?collection_class=ManageIQ::Providers::EmbeddedAnsible::AutomationManager::${type}&attributes=id,name${commonParams}`;
+const catchError = (action) => {
+  window.add_flash(__('Request failed: ') + action, 'error');
+};
+export const loadCatalogs = catalogs => ({
+  type: actionTypes.LOAD_CATALOGS,
+  payload: catalogs,
+});
+export const loadDialogs = () => dispatch => API.get('/api/service_dialogs/?expand=resources&attributes=id,label&sort_by=label&sort_order=ascending').then((data) => {
+  dispatch({
+    type: actionTypes.LOAD_DIALOGS,
+    payload: data,
+  });
+}).catch(() => {
```

Comment:
If you are using manageiq API (API.get etc...), the promise catch will not work like this. In MIQ there is a default catching mechanism. If you want some custom error handling, you must override this and throw the error manually. There is an example: https://github.com/ManageIQ/manageiq-ui-classic/pull/4562/files#diff-1e65f181b5599500c68416e917604382R131